bridge/engine/EngineMixer.cpp

```diff
@@ -1850,14 +1931,33 @@ void EngineMixer::onRtcpPacketDecoded(transport::RtcTransport* sender,
 SsrcOutboundContext* EngineMixer::obtainOutboundSsrcContext(size_t endpointIdHash,
     concurrency::MpmcHashmap32<uint32_t, SsrcOutboundContext>& ssrcOutboundContexts,
     const uint32_t ssrc,
-    const RtpMap& rtpMap)
+    const RtpMap& rtpMap,
+    const bool isForwardSsrc)
 {
     auto ssrcOutboundContextItr = ssrcOutboundContexts.find(ssrc);
     if (ssrcOutboundContextItr != ssrcOutboundContexts.cend())
     {
+        if (ssrcOutboundContextItr->second.markedForDeletion)
+        {
+            assert(isForwardSsrc);
+            return nullptr;
+        }
+
         return &ssrcOutboundContextItr->second;
     }
 
+    if (isForwardSsrc)
```

Comment:
I am not sure about this flag. In a way it is making two methods out of one. We want a normal method to obtain outbound ssrc for rewrite capable streams right? I think it would be cleaner with another obtain method that does the stuff for non-rewrite streams. We want to get rid of them and there are mainly recording streams that use it.